### DISCUSSION TITLE
v52.0-beta1

### DISCUSSION THREAD

**Discussion by westnordost** [streetcomplete/StreetComplete#4888] [Release Discussions]
<img src="https://user-images.githubusercontent.com/4661658/226214144-5a405a98-79d6-4d6c-8759-8739c2c5fc46.png" width="300" align="right"/>

<h3>New overlay for surfaces (#4378, #4642), thanks @matkoniecz</h3>
  <p>Now you can comprehensively map and verify surfaces mapped for roads and paths!</p>
  <h3>New Quests</h3>
  <ul>
  <li><i>Is this still here?</i>, asked for shops every 2 years (#4232, #4843), by @matkoniecz</li>
  </ul>
  <h3>Enhancements and Fixes</h3>
  <ul>
  <li>Addresses: Allow specifying block numbers (#4718, #4850), by @Helium314</li>
  <li>Fix block number input in Japan, by @Helium314</li>
  <li>Address overlay: Addresses can now not be added to buildings in Italy (#4801)</li>
  <li>Address overlay: You can now add addresses to entrances (#4823)</li>
  <li>Surfaces: Added "persistent mud" as selectable surface, by @matkoniecz</li>
  <li>Building levels: Require specifying the roof levels, too (with exceptions) (#4874, #4821, #4661)</li>
  <li>Ask about wheelchair accessibility of toilets in restaurants etc. (#4866, #4868, #4879), thanks @arrival-spring, @Helium314</li>
  <li>You can now duplicate quest presets (#4609, #4820), by @michalgwo</li>
  <li>Fix some memory leaks and crashes (#4869, #4884)</li>
  <li>Other small enhancements (#4864, #4880, #4883...), thanks @matkoniecz, @arrival-spring, @Helium314</li>
  </ul>

<hr /><em>This discussion was created from the release <a href='https://github.com/streetcomplete/StreetComplete/releases/tag/v52.0-beta1'>v52.0-beta1</a>.</em>

**Comment by riQQ** [comment]
*Fix some memory leaks and crashes* mistakenly links to #4887 instead of #4869.

**Reply by FloEdelmann** [reply]
Fixed in https://github.com/streetcomplete/StreetComplete/commit/3d8db797bde5bb998df16ec4d188bb6afd479b60 and manually in the GitHub release description and the discussion description above. Note that it will still show up incorrectly in the changelog in the app itself in v52.0-beta1.

**Comment by westnordost** [comment]
Regarding the attribution, you all probably know that I write `by <person>` when something has been contributed by or `thanks <person>` if that person has had at least a part in its implementation and nothing when it was done by myself only.

It's fine when viewed on GitHub, but since contributions from other people than me became somewhat the majority of the things mentioned in the user-facing changelog, I am concerned that the changelogs become a bit cluttery to read when viewed in the "what's new" dialog within the app. I've been looking into whether the names could be replaced by the github avatars but it's either not possible or some effort (probably possible with [Html.TagHandler](http://developer.android.com/intl/de/reference/android/text/Html.TagHandler.html) + [Html.ImageGetter](http://developer.android.com/intl/de/reference/android/text/Html.ImageGetter.html) + of course adding a build task that downloads the avatars and resizes them to fit) to include in the changelog text in the app.

What do contributors, especially regular contributors like (in no particular order) @matkoniecz , @Helium314, @arrival-spring , @FloEdelmann , @mnalis think of the prospect that within the app, the github names would not be displayed anymore and/or the alternative with the avatar images? Be honest, i.e. speak your mind (you can also send me an email).

**Reply by Helium314** [reply]
I don't care about being mentioned, and I don't think github avatars would be useful or a good replacement because they mean even less than the names for users who aren't active in this repo.

**Reply by matkoniecz** [reply]
It is nice to be credited and vain part of myself likes it.

But while dropping credits may be a bit sad, it also would be fine.

I prefer no credits over github avatar, and it seems that it would make things more cluttered - not less cluttered.

**Reply by FloEdelmann** [reply]
I don't like the idea of replacing the names with avatars: More maintenance work, (slightly) bigger app size, not as self-explanatory as user names, not reducing clutter.

However, I like that people contributing code are also credited in the changelog. IMO, the current changelog is not too cluttered. But if you think it is, another option would be to credit all contributors of a version in one line below all the changes, e.g.:

> Feature ABC
> Fix XYZ
>
> Thanks to @matkoniecz, @Helium314 and @arrival-spring who contributed to this version!

**Reply by westnordost** [reply]
@FloEdelmann That's an interesting suggestion also, but that would mean that also here on GitHub, the contributor names would not be mentioned individually. My thinking was that in the app only, `, (by|thanks) @[a-zA-z0-9_-]+` could be removed and the individual mentions remain in the changelog on GitHub.

**Reply by mnalis** [reply]
I find situation as-is quite ok, and not too cluttered. It's not like we're having hundreds of changes per release.
Credits in app is what is user-facing and thus most rewarding. Changelog in GitHub, or git commit authors is much less externally visible, and thus (while more important from `git blame` perspective) much less important from sociological standpoint, IMHO.

---

I seem to be social animal as much as the next guy -- getting credits, thanks, likes, supportive comments etc. gives me warm fuzzy feelings and pushes me to contribute more to get more of that positive feedback, and seeing credits of others  incites me to contribute too (and helps me to subconsciously navigate virtual social hierarchy).

On retrospect, I think that in early days, devs here (especially @westnordost) acceptance of random newbies and openness to their ideas (even when not in line with his thinking!),  as well as explicitly expressed thanks and graceful mentioning of even the small contributions/contributors is what emitted welcomeness and has drawn me towards this specific community.

So, I think that preserving that is quite more important than tiny amount of clutter in changelog (which is only read once in a while anyway). But I guess I'm probably rationalizing by this point :smile_cat: 

(Also, not a big fan of the avatar idea.)

TL;DR - I'd survive if it vanished, but would be happier if it stayed.

**Reply by HolgerJeromin** [reply]
https://github.com/jens-maus/RaspberryMatic/releases/tag/3.67.10.20230225#:~:text=%F0%9F%91%AA-,Contributors,-(alphabetically)%3A for example also has a Contributors section. I am all-in for Flos suggestion with the one liner.

**Reply by westnordost** [reply]
Okay, so let's leave it as it is

**Comment by rhhsm** [comment]
I've started using the surface overlay and already corrected a few mistakes were asphalt roads were wrongly tagged as sett. I think it would be really useful if `paving_stones` could have a colour different from `asphalt`. Maybe some shade of blue-green or olive?

**Reply by matkoniecz** [reply]
See https://github.com/streetcomplete/StreetComplete/blob/master/app/src/main/java/de/westnordost/streetcomplete/overlays/surface/SurfaceColorMapping.kt and https://github.com/streetcomplete/StreetComplete/blob/master/app/src/main/java/de/westnordost/streetcomplete/overlays/Color.kt

In general, I tried this and failed. If you have good idea which other surfaces should be merged or which specific make to colour palette it would be great to see them.

**Reply by rhhsm** [reply]
Having error finding and maintenance in mind, and sticking to the same colour palette, I think it would be best to group similar surfaces with the same colour. If a road is tagged `concrete` while it should have been tagged with `concrete:lanes`, that's not a big deal, while if it should have been `paving_stones`, it is.

I think assigning colours like this would be an improvement (hope you can easily copy-paste it if acceptable):
    ASPHALT, CHIPSEAL
                       -> Color.BLUE
    CONCRETE, CONCRETE_PLATES, CONCRETE_LANES 
                       -> Color.SKY
    SETT, COBBLESTONE_FLATTENED, UNHEWN_COBBLESTONE, ROCK
                       -> Color.CYAN
    WOOD, METAL, METAL_GRID, GRASS_PAVER, WOODCHIPS -> Color.AQUAMARINE
    PAVING_STONES, PAVING_STONES_WITH_WEIRD_SUFFIX, BRICK, BRICKS
                       -> Color.TEAL
    SAND               -> Color.ORANGE
    GRASS              -> Color.LIME
    DIRT, SOIL, EARTH, MUD, GROUND
                       -> Color.GOLD
    GRAVEL, PEBBLES, COMPACTED, FINE_GRAVEL
                       -> Color.GRAY
    CLAY, ARTIFICIAL_TURF, TARTAN
                       -> Color.BLACK // not encountered in normal situations, get the same as surface with surface:note
    UNKNOWN, PAVED, UNPAVED
                       -> Color.BLACK

 It can probably be improved more, for instance we might want to use gray for the concrete instead of the gravel group, and orange for surfaces that need checking (because they are unusual and possibly wrong), so we could move SAND to the DIRT, SOIL, EARTH, MUD, GROUND group, and make PAVING_STONES_WITH_WEIRD_SUFFIX orange.
WOOD, METAL, METAL_GRID are almost always bridge or steps surfaces, in my experience.

**Reply by westnordost** [reply]
Use three backticks (```) at the start and end of blocks of code to make them be formatted and highlighted as code. In this case, it would be most helpful if you made github hightlight your suggestion as a diff. For that, use "diff", e.g.
<pre>
```diff
CLAY, ARTIFICIAL_TURF, TARTAN
- -> Color.BLACK
+ -> Color.BLUE_PINK_CHEQUERED
```
</pre>
is displayed as
```diff
CLAY, ARTIFICIAL_TURF, TARTAN
- -> Color.BLACK
+ -> Color.BLUE_PINK_CHEQUERED
```

**Reply by matkoniecz** [reply]
> move SAND to the DIRT, SOIL, EARTH, MUD, GROUND group

In countryside, at least for cyclist, sand is massively different from earth. In fact, one of my main motivations to implement it was mapping paths and roads unycleable due to `surface=sand` that is not yet mapped.

**Reply by rhhsm** [reply]
@matkoniecz I see your point. We should try to imagine which surfaces have important differences for the user, and use different colours for them, while for surfaces that are not very different for the user and that we don't have enough colours for can be the same colour.

**Reply by westnordost** [reply]
I've used the new surface overlay myself and somewhat missed a visual distinction between paving stones and asphalt. Curiously, concrete plates are colored the same way as sett. What about just giving paving stones the sett-color?

**Reply by rhhsm** [reply]
I'd like to see different colours for paving stones and sett. Considering so many wishes, would it be an option of the user could choose the colours by themselves?

**Reply by westnordost** [reply]
No

**Reply by westnordost** [reply]
What about if asphalt would get the same color as "other", i.e. just black?

**Reply by matkoniecz** [reply]
That results in a map looking really ugly and depressing in cities (at least it was when asphalt was gray - I cannot imagine black being any better), though maybe "other" can take some other color.

I guess that it results in losing ability to spot mistagged  CLAY, ARTIFICIAL_TURF, TARTAN - but how often it happens?

**Reply by westnordost** [reply]
https://github.com/streetcomplete/StreetComplete/pull/4946
